content/en/docs/contribution-guidelines/upgrade-graphs.md

```diff
@@ -0,0 +1,234 @@
+---
+title: Upgrade Graphs Diagrams
+linkTitle: Upgrade Graphs
+---
+
+## Standards and definitions
+
+| <div style="align-content: center;">Description</div>   | <div style="align-content: center;width:200px">Diagram Example</div>       | 
+|----------|:-------------:|
+| bundle which is the head of the channel | {{<mermaid>}}
+flowchart TB
+   classDef head fill:#ffbfcf;
+   classDef installed fill:#34ebba;
+   
+   ID(v0.0.1):::head
+{{</mermaid>}} |
+| bundle which is the installed currently | {{<mermaid>}}
+flowchart TB
+   classDef head fill:#ffbfcf;
+   classDef installed fill:#34ebba;
+   
+   ID(v0.0.1):::installed
+{{</mermaid>}} |
+| bundle which can be installed | {{<mermaid>}}
+flowchart TB
+   classDef head fill:#ffbfcf;
+   classDef installed fill:#34ebba;
+   
+   ID(v0.0.1)
+{{</mermaid>}} |
+| channel | {{<mermaid>}}
+flowchart TB
+   classDef head fill:#ffbfcf;
+   classDef installed fill:#34ebba;
+   
+   subgraph preview
+      ID(v0.0.1) 
+   end
+{{</mermaid>}} |
+| default channel | {{<mermaid>}}
+flowchart TB
+   classDef head fill:#ffbfcf;
+   classDef installed fill:#34ebba;
+   
+   subgraph "preview (default)"
+      ID(v0.0.1) --> ID2(v0.0.2)
+   end
+{{</mermaid>}} |
+| replaces action to specify an upgrade path for a project which will replace one version for another. It is defined in the CSV. (eg.`spec.replaces: exampleoperator.v0.0.1`) | {{<mermaid>}}
+flowchart TB
+   classDef head fill:#ffbfcf;
+   classDef installed fill:#34ebba;
+   
+   A(v0.0.1) --> B(v0.0.4)
+{{</mermaid>}} |
+| skips action to skip one or more specific project versions (1..N). It is defined in the CSV (eg.`spec.skips: exampleoperator.v0.0.2, exampleoperator.v0.0.3`)| {{<mermaid>}}
+flowchart TB
+   classDef head fill:#ffbfcf;
+   classDef installed fill:#34ebba;
+   
+   A(v0.0.1) x--x |v0.0.2,v0.0.3| B(v0.0.4)
+{{</mermaid>}} |
+| skipRange action which specify a range that should not be used to upgrade a specific version to another. It is defined in the CSV. (E.g `annotations.olm.skipRange: '>= 2.0.4 < 3.0.0'`) | {{<mermaid>}}
+flowchart TB
+   classDef head fill:#ffbfcf;
+   classDef installed fill:#34ebba;
+   
+   C(v2.0.3) o--o |>= 2.0.4 < 3.0.0| D(v3.0.1)
+{{</mermaid>}} |
+| future replace | {{<mermaid>}}
+flowchart TB
+   classDef head fill:#ffbfcf;
+   classDef installed fill:#34ebba;
+   
+   A(v0.0.1) -.-> B(v0.0.4)
+{{</mermaid>}} |
+| future skips | {{<mermaid>}}
```

Comment:
Could you explain a little more on what you mean by `future` skips/replaces etc? Not sure where we'll need to use those.

Reply:
The future actions are required to describe scenarios where we would like to represent future graph paths. For example, if we need to speak/describe how they can work with CEVs. Then we need to create a diagram with the current scenario and we can demonstrate how it will be in the future.  

Another good case scenario is if we need to explain changes such as for the next future oLM versions the channel A would use the label and how the future versions will be.